MuvrKit/MKWatchConnectivity.swift

```diff
@@ -249,20 +249,17 @@ public final class MKConnectivity : NSObject, WCSessionDelegate {
 
             // update the number of recorded samples
             let readFromDate = NSDate(timeIntervalSince1970: sensorData.end)
-            let updatedProps = props.with(accelerometerEnd: readFromDate)
-            self.sessions[session] = updatedProps
+            if let p = self.sessions[session] {
```

Comment:
Correct, this one is probably fine.
I think the problem lies in the transfer data callback